plugins/org.eclipse.emf.ecoretools.ale.core/src/org/eclipse/emf/ecoretools/ale/core/env/impl/DslConfiguration.java

```diff
@@ -0,0 +1,213 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Inria and Obeo.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Inria - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.emf.ecoretools.ale.core.env.impl;
+
+import static java.lang.String.join;
+import static java.util.Arrays.stream;
+import static java.util.stream.Collectors.toList;
+import static org.eclipse.emf.ecoretools.ale.core.io.IOResources.toFile;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.emf.ecoretools.ale.core.env.IAleEnvironment;
+import org.eclipse.emf.ecoretools.ale.core.io.IOResources;
+
+/**
+ * An {@link IAleEnvironment ALE environment} that is defined as
+ * a {@link Properties property file}.
+ * <p>
+ * This class is not supposed to be instantiated by clients, call 
+ * {@link IAleEnvironment#fromPaths(Collection, Collection)}
+ * instead.
+ */
+// TODO [API] Should provide a load() method instead of loading
+//		      the .dsl file in the constructor
+public final class DslConfiguration extends AbstractAleEnvironment {
```

Comment:
For consistency and clarification with other implementation of _AbstractAleEnvironment_, I suggest that this file should be named _DslConfigurationAleEnvironment_

Reply:
I'll be picky, but `DslConfigurationAleEnvironment` looks odd to me: I am really bothered by those "artificial" suffixes. I actually though of `DslConfigurationFileBasedAleEnvironment` which is the more explicit name I could found. I finally opted for `DslConfiguration` because:
 - instances of the class represent DSL configuration files (`DslConfigurationFile` would thus have been a better name)
 - `DslConfigurationFileBasedAleEnvironment`  is really verbose
 - users are not supposed to instantiate the class directly

Given the following code:
```java
try (DslConfiguration dsl = IAleEnvironment.fromDslFile(new File("fsm.dsl"))) {

}
```
- we understand that `DslConfiguration` is an ALE environment
- `DslConfigurationAleEnvironment` would make (IMO) the try-with-resource statement way less readable

What do you think?

Reply:
This is obviously a matter of preference for name convention. 
I'm may old-fashioned but I usually prefer class names that are as much as possible self-explanatory about the "what" they are (a class defines the type of the object)
This is quite useful for example when debugging and you obtain an instance of "Normalized" or DslConfiguration,  with such short name and without fine understanding of the code, you have to open the declaration to understand that its main role is to provide an AleEnvironnment


you're right that at the level of the constructor, the information is redundant but it isn't available anymore when using the dsl variable  (is in the rest of the code), the hovering will tell you that this is a DslConfiguration, but you need to navigate in the code to get it main goal

So most of the time I prefer longer names than short name that force to navigate in the code to understand it.

In situations where we have a lot of chained decorators, and if the user has regularly to use them in chain then I agree that shorter names could be useful  (I have examples in mind with an API to construct expressions for example) but in our case I believe that the construction doesn't involve a long chain of constructors 

That said, I can live with it if all classes use the same name convention ;-)

Reply:
Renamed as `FileBasedAleEnvironment`. I believe the name is comprehensive enough to understand and abstract enough to allow us to file formats later if needed.